node/blockAPI/shardBlock.go

```diff
@@ -0,0 +1,99 @@
+package blockAPI
+
+import (
+	"encoding/hex"
+
+	apiBlock "github.com/ElrondNetwork/elrond-go/api/block"
+	"github.com/ElrondNetwork/elrond-go/data/block"
+	"github.com/ElrondNetwork/elrond-go/dataRetriever"
+)
+
+type shardAPIBlockProcessor struct {
+	*baseAPIBockProcessor
+}
+
+// NewShardApiBlockProcessor will create a new instance of shard api block processor
+func NewShardApiBlockProcessor(arg *APIBlockProcessorArg) *shardAPIBlockProcessor {
+	isFullHistoryNode := arg.HistoryProc.IsEnabled()
+	return &shardAPIBlockProcessor{
+		baseAPIBockProcessor: &baseAPIBockProcessor{
+			isFullHistoryNode:        isFullHistoryNode,
+			selfShardID:              arg.SelfShardID,
+			store:                    arg.Store,
+			marshalizer:              arg.Marshalizer,
+			uint64ByteSliceConverter: arg.Uint64ByteSliceConverter,
+			historyProc:              arg.HistoryProc,
+			unmarshalTx:              arg.UnmarshalTx,
+		},
+	}
+}
+
+// GetBlockByNonce will return a shard APIBlock by nonce
+func (sbp *shardAPIBlockProcessor) GetBlockByNonce(nonce uint64, withTxs bool) (*apiBlock.APIBlock, error) {
+	storerUnit := dataRetriever.ShardHdrNonceHashDataUnit + dataRetriever.UnitType(sbp.selfShardID)
+
+	nonceToByteSlice := sbp.uint64ByteSliceConverter.ToByteSlice(nonce)
+	headerHash, err := sbp.store.Get(storerUnit, nonceToByteSlice)
+	if err != nil {
+		return nil, err
+	}
+
+	blockBytes, err := sbp.getFromStorer(dataRetriever.BlockHeaderUnit, headerHash)
+	if err != nil {
+		return nil, err
+	}
+
+	return sbp.convertShardBlockBytesToAPIBlock(headerHash, blockBytes, withTxs)
+}
+
+// GetBlockByHash will return a shard APIBlock by hash
+func (sbp *shardAPIBlockProcessor) GetBlockByHash(hash []byte, withTxs bool) (*apiBlock.APIBlock, error) {
+	blockBytes, err := sbp.getFromStorer(dataRetriever.BlockHeaderUnit, hash)
+	if err != nil {
+		return nil, err
+	}
+
+	return sbp.convertShardBlockBytesToAPIBlock(hash, blockBytes, withTxs)
+}
+
+func (sbp *shardAPIBlockProcessor) convertShardBlockBytesToAPIBlock(hash []byte, blockBytes []byte, withTxs bool) (*apiBlock.APIBlock, error) {
+	blockHeader := &block.Header{}
+	err := sbp.marshalizer.Unmarshal(blockHeader, blockBytes)
+	if err != nil {
+		return nil, err
+	}
+
+	headerEpoch := blockHeader.Epoch
+
+	numOfTxs := uint32(0)
+	miniblocks := make([]*apiBlock.APIMiniBlock, 0)
+	for _, mb := range blockHeader.MiniBlockHeaders {
+		if mb.Type == block.PeerBlock {
```

Comment:
Is this the only case when we want to skip? (just to confirm)

Reply:
yes. because we don't want return peer miniblocks. this skip is did also at elasticsearch indexer.